morpheus/stages/input/in_memory_data_generation_stage.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2022-2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any
+from typing import Callable
+from typing import Generator
+from typing import Type
+
+import mrc
+
+from morpheus.config import Config
+from morpheus.pipeline.single_output_source import SingleOutputSource
+from morpheus.pipeline.stage_schema import StageSchema
+
+logger = logging.getLogger(f"morpheus.{__name__}")
+
+
+class InMemoryDataGenStage(SingleOutputSource):
```

Comment:
I think this could be made to be more generic. Then you could support any sort of iterable. Not just generator functions.

Reply:
Also, make the `InMemorySource` stage then derive from this.